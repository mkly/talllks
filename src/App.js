import { useState } from "preact/hooks";
import { html } from "htm/preact";

import "normalize.css/normalize.css";

import { css } from "emotion";

import cows from "./Data/cows";

import Cow from "./Component/Cow";
import Show from "./Component/Show";
import isVisible from "./HOC/isVisible";

const ShowVisible = isVisible(Show);
const CowVisible = isVisible(Cow);

const mainStyle = css`
  width: 100%;
  margin: 0 auto;
  height: 100%;
  display: flex;
  align-items: center;
  justify-content: center;
  flex-direction: column;
`;

const App = () => {
  const [cowVisible, setCowVisible] = useState(false);
  const [currentCow, setCurrentCow] = useState(cows[0]);

  const showCowHello = () => setCowVisible(true);
  const showRandomCowHello = () => {
    let cow = cows[Math.floor(Math.random() * cows.length)];
    while (cow == currentCow) {
      cow = cows[Math.floor(Math.random() * cows.length)];
    }
    setCurrentCow(cow);
  };

  return html`
    <main class=${mainStyle}>
      <${CowVisible} visible=${cowVisible} cow=${currentCow} />
      <${ShowVisible}
        visible=${!cowVisible}
        showCowHello=${showCowHello}
        text="Say Hello!"
      />
      <${ShowVisible}
        visible=${cowVisible}
        showCowHello=${showRandomCowHello}
        text="Say a Random Hello!"
      />
    </main>
  `;
};

export default App;
