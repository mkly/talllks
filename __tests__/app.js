/**
 * @jest-environment jsdom
 */
import { html, render } from "htm/preact";

import App from "../src/App";

describe("App", () => {
  it("Should display App", () => {
    const body = document.createElement("div");
    document.body.appendChild(body);

    render(html`<${App}/>`, body);
    expect(body.innerHTML).toMatchSnapshot();
  });
});
