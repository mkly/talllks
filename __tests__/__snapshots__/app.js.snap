// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Should display App 1`] = `"<main class=\\"css-1w8y0mw\\"><div class=\\"css-1qqjzai\\"><button class=\\"css-1qfxjdt\\">Say Hello!</button></div></main>"`;
